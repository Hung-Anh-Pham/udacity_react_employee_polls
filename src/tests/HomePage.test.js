import { MemoryRouter } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import HomePage from "../components/HomePage";
import { questions } from "../utils/_DATA";
import { renderWithProviders } from "./setup-test";


const mockStore = configureMockStore();

describe("Test HomePage component", () => {
	let store;

	beforeEach(() => {
		const initialState = {
			authedUser: {
				id: "tylermcginnis",
				password: "abc321",
				name: "Tyler McGinnis",
			},
			questions: questions,
		};
		store = mockStore(initialState);
	});

	it("Should render correct QuestionCard component", () => {

		const { page } = renderWithProviders(
			<MemoryRouter>
				<HomePage />
			</MemoryRouter>,
			{ store }
		)

		expect(page).toMatchSnapshot();
	});
});
