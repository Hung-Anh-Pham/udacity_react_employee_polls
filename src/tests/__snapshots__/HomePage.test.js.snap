// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HomePage component Should render correct QuestionCard component 1`] = `undefined`;
